src/fides/api/api/v1/endpoints/system.py

```diff
@@ -134,10 +134,15 @@ def patch_connections(
     status_code=HTTP_204_NO_CONTENT,
     response_model=None,
 )
-def delete_connection(
-    connection_key: FidesKey, *, db: Session = Depends(deps.get_db)
-) -> None:
-    delete_connection_config(db, connection_key)
+def delete_connection(fides_key: str, *, db: Session = Depends(deps.get_db)) -> None:
+    system = get_system(db, fides_key)
+    if system.connection_configs is None:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail="No integration found linked to this system",
+        )
+
+    delete_connection_config(db, system.connection_configs.key)
```

Comment:
This threw me off a bit but I see that `connection_configs` will be a single connection config (for now). I think it's worth adding a comment to clarify this. Also, are you removing the need for a connection key since a systems only have one connection right now?